fileio/ft_write_data.m

```diff
@@ -875,12 +875,97 @@ function ft_write_data(filename, dat, varargin)
     data.label = hdr.label;
     data.sampleinfo = [1 size(dat,2)];
     
-    % convert the raw data structure to Homer format
-    nirs = fieldtrip2homer(data, 'event', evt);
+    % devide data in nirs channels, stimulus channels and auxillary channels
+    seldat  = startsWith(hdr.chantype, 'nirs');
+    selstim = strcmp(hdr.chantype, 'stimulus');
+    selaux  = ~seldat & ~selstim;
+    
+    % create empty SnirfClass
+    snirf = SnirfClass();
+    
+    % collect information for creation of snirf file
+    source_idx=find(contains(data.hdr.opto.optotype, {'transmitter', 'source'}));
+    detector_idx=find(contains(data.hdr.opto.optotype, {'receiver', 'detector'}));
+    tra=data.hdr.opto.tra;
+    all_wavelengths=data.hdr.opto.wavelength(tra(find(tra>0)));
+    split=median(all_wavelengths);
+    WL1.values=all_wavelengths(all_wavelengths<split);
+    WL2.values=all_wavelengths(all_wavelengths>split);
+    WL1.nominal=round(median(WL1.values),-1);
+    WL2.nominal=round(median(WL2.values),-1);
+    num_WL=2;
+    ft_warning('Fieldtrip assumes that nominal wavelengths of %d and %d nm were used. Please adjust manually when this is not correct', WL1.nominal, WL2.nominal)
+     
+    % metaDataTags
+    snirf.metaDataTags(1).tags.LengthUnit = data.hdr.opto.unit; 
+    snirf.metaDataTags(1).tags.TimeUnit = 's'; 
+    snirf.metaDataTags(1).tags.FrequencyUnit = 'Hz';
+    
+    % data 
+    % (FIX ME: in theory this can contain multiple data blocks)
+    snirf.data(1).dataTimeSeries = data.trial{1}(seldat,:)'; % <number of time points> x <number of channels>
+    snirf.data(1).time = data.time{1}'; % <number of time points x 1> (can also be  represented as <start time x sample time spacing>
+    
+    % measurementList
+    for i=1:size(tra,1)
+      source=find(tra(i,:)>0);
+      detector=find(tra(i,:)<0);
+      snirf.data.measurementList(i).sourceIndex = find(source_idx==source);
+      snirf.data.measurementList(i).detectorIndex = find(detector_idx==detector);
+%       snirf.data.measurementList(i).wavelengthActual =
+%       all_wavelengths(i); % is not yet supported by the snirf toolbox
+      if any(all_wavelengths(i)==WL1.values)
+        snirf.data.measurementList(i).wavelengthIndex = 1;
+      else
+        snirf.data.measurementList(i).wavelengthIndex = 2;
+      end
+      snirf.data.measurementList(i).dataType = 99999;
+      snirf.data.measurementList(i).dataTypeLabel = 'dOD';
+    end
+    ft_warning('FT assumes that the input data is change in optical density. Adjust manually if not correct')
```

Comment:
can you do 
    ft_warning('assuming that the input data represents (change in) optical densities')

The "Adjust manually" instruction is not specific enough (i.e. I would not know what to do if that were to appear on screen), so I have left that out. If you want to keep that, you should elaborate. And in that case it wou8ld probably be better do link to https://www.fieldtriptoolbox.org/getting_started/snirf/#writing-snirf-data and extend the help there rather than in the warning message that appears in the MATLAB the command window.